module/zfs/zfs_ioctl.c

```diff
@@ -4285,8 +4306,8 @@ zfs_ioc_recv_impl(char *tofs, char *tosnap, char *origin,
 	/*
 	 * On error, restore the original props.
```

Comment:
I think we should, i'm still kind of struggling to come up with a way to stash away all existing properties (recvd, local and inherited) in a sane way that would not involve too many different nvlists and loops on restore.